lib/matplotlib/tests/test_image.py

```diff
@@ -603,7 +603,8 @@ def test_load_from_url():
 
 @image_comparison(baseline_images=['log_scale_image'],
                   remove_text=True)
-def test_log_scale_image():
+# The recwarn fixture captures a warning in image_comparison.
+def test_log_scale_image(recwarn):
```

Comment:
I think this is throwing a warning that we should try and fix instead of ignoring it. Presumably `ax.imshow` sets `axis=equal` behind the scenes, which causes `ax.set_yscale('log')` to throw a warning. If this is all valid code (as I think it is) I don't think it should throw a warning.

Reply:
But how could we catch it? Setting equal aspect ratio and setting log scaling are independent, and I _think_ there's no link saying that equal aspect ratio was requested by `imshow`.

Reply:
In principle we could set the aspect to 'auto' after calling `imshow` but I would expect that to change the output.  I am :+1: on just catching this warning.